### PR TITLE
use past hidden states

### PR DESCRIPTION
**Pull request by AccidentallyOnPurpose**

### CODE REVIEW

File: generator/gpt2/gpt2_generator.py [hunk 1]
```diff
@@ -16,6 +15,42 @@
 logging.getLogger("transformers.modeling_utils").setLevel(logging.WARN)
 logging.getLogger("transformers.configuration_utils").setLevel(logging.WARN)
 
+# This will be in the next transformers release, delete then
+def top_k_top_p_filtering(logits, top_k=0, top_p=1.0, filter_value=-float("Inf"), min_tokens_to_keep=1):
```
Comment (thadunge2) [inline generator/gpt2/gpt2_generator.py]: Shouldn't this be inside the model class?

**Review by thadunge2 — CHANGES_REQUESTED** [review]

File: generator/gpt2/gpt2_generator.py (hunk 1 above)
Comment (AccidentallyOnPurpose) [reply generator/gpt2/gpt2_generator.py]: Which model class? Do you mean inside transformers.model_utils?

If you mean model_utils

The thing is it's inside there in the latest git commit of transformers. So it's missing in the latest version - 2.3.0. We would have to install a specific git commit in order to import it.

If you mean in the GPT2 class. No, neither here nor in the transformers is it as a method on the GPT2 model class. It's doesn't require anything from self.

Comment (thadunge2) [reply generator/gpt2/gpt2_generator.py]: But it's called as self.top_k_top_p_filtering in the generate method.

Comment (thadunge2) [reply generator/gpt2/gpt2_generator.py]: Whoops, nevermind, I was thinking of something else!

**Pull request merged**